libass/ass_directwrite.c

```diff
@@ -253,79 +253,90 @@ static void init_FallbackLogTextRenderer(FallbackLogTextRenderer *r,
     };
 }
 
+/*
+ * A helper function for init_font_private_stream and get_utf8_path.
+ * Remember to release the file and loader after obtaining it, even if the function fails.
+ */
+static bool get_font_file_loader_and_reference_key(FontPrivate *priv,
+                                                   const void **refKey,
+                                                   UINT32 *keySize,
+                                                   IDWriteFontFile **file,
+                                                   IDWriteFontFileLoader **loader)
+{
+    HRESULT hr = S_OK;
+    UINT32 n_files = 1;
+    *refKey = NULL, *file = NULL, *loader = NULL, *keySize = 0;
```

Comment:
I think it's better to not abuse operator `,` in this context. Also I think it's more logically correct to move this initialization at the very start of the function, as `hr` and `n_files` are directly related to `IDWriteFontFace_GetFiles` call.

Reply:
I could move it to the start. I used the operator `,` because I didn't really want to make it too verbose by occupying four lines and then a newline for better visual clarity... (cause we weren't really sure if that nulling is needed to begin with, it was done just to have the function output behavior be cleaner no matter what comes in the args in.